include/mender-storage.h

```diff
@@ -27,6 +27,11 @@ extern "C" {
 #include "mender-utils.h"
 #include "mender-update-module.h"
 
+/**
+ * @brief Deployment data version number
+ */
+#define MENDER_STORAGE_DEPLOYMENT_DATA_VERSION 1
```

Comment:
I know I [said](https://github.com/mendersoftware/mender-mcu/pull/39#discussion_r1743399477) to put it in this file, but I meant is "as encapsulated as possible", which in this PR means `mender-storage.c`. Makes sense? At the time as we were discussing in the `mender-client.c` I only thought about header files